generator/src/test/java/com/scottlogic/deg/generator/outputs/targets/FileOutputTargetTests.java

```diff
@@ -0,0 +1,86 @@
+package com.scottlogic.deg.generator.outputs.targets;
+
+import com.scottlogic.deg.common.profile.Profile;
+import com.scottlogic.deg.generator.outputs.formats.OutputFormat;
+import com.scottlogic.deg.generator.utils.FileUtils;
+import org.junit.jupiter.api.Test;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class FileOutputTargetTests {
+    //Create Mocks
+    private Path mockFilePath = mock(Path.class);
+    private FileUtils mockFileUtils = mock(FileUtils.class);
+    private OutputFormat mockOutputFormat = mock(OutputFormat.class);
+    private Path mockParentPath = mock(Path.class);
+
+    @Test
+    public void validate_generateOutputFileIsADirectory_throwsException(){
+        //Arrange
+        when(mockFileUtils.isDirectory(any())).thenReturn(true);
+        when(mockFilePath.getParent()).thenReturn(mockParentPath);
+        FileOutputTarget outputTarget = new FileOutputTarget(mockFilePath, mockOutputFormat, false, mockFileUtils);
+
+        //Act
+        OutputTargetValidationException thrown = assertThrows(OutputTargetValidationException.class, ()->outputTarget.validate(),"Expected OutputTargetValidationException to throw, but didn't");
+
+        //Assert
+        assertEquals( "target is a directory, please use a different output filename", thrown.getMessage());
```

Comment:
I'm not convinced that testing the exact wording of the exception message is a good idea.  In theory the content of the exception message should be almost entirely cosmetic, so I don't think changing it should break the test.  Definitely test that the exception is thrown; what I might do as well is test that, for example, the message is not empty (which would pass) or that the message contains the name of the failing target (which the current code would fail!).

Again this is something that other people may have different opinions on